cli/pcluster/config/config_patch.py

```diff
@@ -0,0 +1,296 @@
+        check that all sections have been correctly matched by just comparing their labels.
+        """
+        for section_key in self.base_config.get_section_keys():
+            i = 1
+
+            for _, base_section in sorted(self.base_config.get_sections(section_key).items()):
+
+                new_lbl = "_{0}{1}".format(
+                    section_key, i if i > 1 else ""  # Avoid number if first or unique section per key
+                )
+
+                # Make sure the new label is not present in target config
+                while self.target_config.get_section(section_key, new_lbl):
+                    new_lbl = "_" + new_lbl
+
+                base_section.label = new_lbl
+                i += 1
+
+    @property
+    def update_policy_level(self):
+        """Get the max update policy level of the ConfigPatch."""
+        return (
+            max(change.update_policy.level for change in self.changes)
+            if len(self.changes)
+            else UpdatePolicy.ALLOWED.level
+        )
+
+    def _create_default_section(self, config, section):
+        """Create a default section of same type of the provided section in the provided config to allow comparison."""
+        section_definition = section.definition
+        section_type = section_definition.get("type")
+        mock_section = section_type(
+            section_definition=section_definition, pcluster_config=self.target_config, section_label=section.label
+        )
+        # Sections with key param (like EBS) are marked as mock to be ignored from related settings params
+        if section.key_param:
+            mock_section.mock = True
+        config.add_section(mock_section)
+        return mock_section
+
+    def _get_target_section_by_param(self, section_key, param_key, param_value):
+        section = None
+        sections = self.target_config.get_sections(section_key)
+        for _, s in sections.items():
+            param = s.get_param(param_key)
+            if param and param.value == param_value:
+                section = s
+                break
+        return section
+
+    def _get_target_section(self, base_section):
+        section = None
+        key_param = base_section.key_param
+        if key_param:
+            # EBS sections are looked by shared dir
+            key_param_value = base_section.get_param_value(key_param)
+            section = self._get_target_section_by_param(base_section.key, key_param, key_param_value)
+        else:
+            # All other sections are matched just by key
+            section = self.target_config.get_section(base_section.key)
+
+        # If section is not present in target config we build a default section to allow comparison
+        if not section:
+            section = self._create_default_section(self.target_config, base_section)
+
+        return section
+
+    def _get_base_section(self, target_section):
```

Comment:
They are a bit different in purpose and logic: in target the match is "semantic" (by key or  primary_key), while in base is by label, since all sections should be already matched at that point.